### ISSUE TITLE
Subject: Inquiry about Backbone Replacement in Your Project

### ISSUE THREAD

**Issue by hjmbzr8nvp-cyber** [blaz-r/SuperSimpleNet#20]
Dear Author,

Your project is truly impressive! However, I have encountered a few issues during the reproduction process and I was wondering if you could kindly help clarify my doubts. The backbone network used in the source code is wide_resnet50_2. Have you implemented any other models with a smaller number of parameters? If I were to switch to a different backbone, would it be necessary to adjust other hyperparameter settings? If so, which parameters should be tuned to properly adapt to the new backbone? I look forward to your reply. Thank you very much for your time.

**Comment by blaz-r** [comment]
Hey, you can replace the backbone without modification of any hyperparameters for some of the backbones, but some **will** require adjustment.
These were validated and work with same setup (though performance-wise WRN50 is best):
- "resnet101"
- "resnet50"
- "wide_resnet101_2"

"resnet18" also works, but requires more epochs if I remember correctly. Maybe with IN1k V1 weights, even more models (that have about the same ResNet-ish structure and batchnorm layers) will maybe work.

What you need to be careful about is that the feature extractor uses Imagenet V1 weights ("IMAGENET1K_V1" which is the default in the code). Even using the same WRN50 but V2 weights will not work, as the magnitude of features goes up by about an order of magnitude. In my limited testing, this can be fixed by also increasing std, but I'd advise you stick to V1.
